### DISCUSSION TITLE
Temporarily bypass frozen field in pydantic model

### DISCUSSION THREAD

**Discussion by dvanhende-oqc** [pydantic/pydantic#10814] [Question]
Hi everyone,

I am currently implementing a pydantic model that has a frozen field. However, for a specific use case, I want to bypass this frozen flag to adjust a field. My current minimal example (see below) gives an error saying that my field `a` is still frozen even after I have set the `frozen` flag to False.

How can I make this code work without changing the `frozen=True` flag in `a`?

```py
from pydantic import BaseModel, Field

class TestModel(BaseModel):
    a: int = Field(frozen=True)
    
tm = TestModel(a=1)

tm.model_fields["a"].frozen = False

tm.__pydantic_validator__.validate_assignment(tm, "a", 5)
```

**Comment by Viicos** [comment]
If you don't care about validation, you can edit the `__dict__` directly:

```python
tm.__dict__['a'] = 5
```

**Reply by dvanhende-oqc** [reply]
Thanks! And if I do care about validation?

**Reply by Viicos** [reply]
Temporarily switching the frozen behavior of a field is unusual, so we don't really support this use case. One generic solution that could be implemented could be with a context manager:

```python
with tm.temp_change_field_config('a', frozen=False):  # API and name TBD
    tm.a = 5
```

but requires a substantial amount of work and I don't think this will happen in the near future.

The reason why `tm.model_fields["a"].frozen = False` doesn't work is because internally, when you define your model, we build what we call a _core schema_, that is used to build the pydantic-core validator (accessible through the `__pydantic_validator__` attribute). Changing anything in `models_fields` won't trigger an update of the core schema and the validator.

**Reply by dvanhende-oqc** [reply]
Thanks for the help @Viicos !
